Jobs APIupdate + departments grouping

fairlighteth opened the pull request:
# Summary

- Pulls all jobs data from a new public `cowswap` specific Greenhouse endpoint
- Segments and displays jobs by departments (before we didn't have this data)
- Allows to filter jobs by a specific departments + shows the count

https://user-images.githubusercontent.com/31534717/162991628-1766e45f-079c-4f64-86ad-f1b38cb8d073.mov

No jobs:

<img width="1437" alt="Screen Shot 2022-04-12 at 15 52 03" src="https://user-images.githubusercontent.com/31534717/162991686-1f645bd8-4b1b-4202-82ba-698d532371ea.png">

elena-zh commented:
Hey @fairlighteth , great changes!
However, seems that I have issues with the font in the dropdown: it looks different. 
![font](https://user-images.githubusercontent.com/70885163/163037949-f98e3d2e-12f4-4793-9c53-bfab98c2613b.jpg)
![font1](https://user-images.githubusercontent.com/70885163/163037959-5e6cecd9-0841-4b75-ae9f-008ad287acca.jpg)

Besides, the dropdown itself is in white and looks different in comparison with your video above
![in black](https://user-images.githubusercontent.com/70885163/163038183-2ec07af5-1a1a-4aac-84d0-918d03003f44.jpg)

And a nitpick: could we please increase the margin between these lines? 
![image](https://user-images.githubusercontent.com/70885163/163038828-8a45bc2b-ce55-4a57-9076-f4ca82e3c80f.png)

Could you please take a look into these issues?

Thanks!

alfetopito commented on pages/jobs.tsx:
```diff
@@ -1,29 +1,39 @@
+  try {
+    const response = await fetch(api)
+    const data = await response.json()
+    data.jobs.map(job => {
```
You are not returning anything from the `map`, so a better function for this is `forEach`

fairlighteth replied:
Ah right, I'm not storing the return value of the `.map` in a variable. Updated and thanks for reviewing.

fairlighteth commented:
@elena-zh Addressed your comments. One thing that I haven't been able to change, is the grey hover background color for Windows. I'd suggest to leave as is for now.

<img width="1249" alt="Screen Shot 2022-04-13 at 11 47 54" src="https://user-images.githubusercontent.com/31534717/163164593-fe01d36b-aed7-4544-9c68-8cc88c872121.png">

alfetopito reviewed (approved):
Looks awesome!

Only one minor nitpick comment

elena-zh reviewed (approved):
@fairlighteth , looks great now!
Thank you!
